tsup.config.ts

```diff
@@ -1,14 +1,22 @@
 import { defineConfig } from 'tsup';
 
-export default defineConfig({
-  entry: {
-    index: 'lib/index.ts',
-    'v4/index': 'lib/v4/index.ts',
-  },
-  splitting: false,
-  sourcemap: true,
-  clean: true,
-  format: ['cjs', 'esm'],
-  dts: true,
-  outDir: 'dist',
-});
+export default [
+  defineConfig({
+    entry: ['lib/v3/index.ts'],
+    splitting: false,
+    sourcemap: true,
+    clean: true,
+    format: ['cjs', 'esm'],
+    dts: true,
+    outDir: 'v3',
+  }),
+  defineConfig({
+    entry: ['lib/v4/index.ts'],
+    splitting: false,
+    sourcemap: true,
+    clean: true,
+    format: ['cjs', 'esm'],
+    dts: true,
+    outDir: 'v4',
+  }),
+];
```

Comment:
dev: here lies the gist of this PR.

Instead of building files under `./dist`, we build under `./v3` and `./v4` respectively, thus allowing typescript to resolve `zod-validation-error/v4` without using the package.json exports declaration (because it can rely on the path).

Reply:
No, I mean having the dist `v3` and `v4` outside a `dist` or `build` directory

Reply:
Great, thanks for the clarifying 👍